docs: Add guides example about #423

leestarb opened the pull request:
## What type of pull request is this?

<!-- Check whichever applies to your PR -->
- [ ] Non-breaking code change
- [ ] Breaking code change
- [x] Documentation change/addition
- [ ] Tests change

## Description
<!-- Clearly and concisely describe what this PR is for, and why you feel it should be merged. -->
!

## Changes
<!-- - A bullet pointed list outlining the changes you have made -->
- Add "Scale switching" guides example

## Checklist

<!-- These are actions you **must** have taken, if you haven't, your PR will be rejected -->
- [ ] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [ ] I've tested my code

leestarb committed:
docs: Add guides example about #423

LordOfPolls commented on docs/src/Guides/20 Scales.md:
````diff
@@ -203,6 +203,33 @@ bot.grow_scale("Filename_here")
 bot.start("token")
 ```
 
+Use also can pass any **keyword** arguments into the `setup` function. Here is a basic "Scale switching" example:
````
Grammatical Error. Do you mean `you can also pass` or `users can also pass`?

leestarb committed:
docs: Correct grammar

leestarb committed:
docs: Improve Scales guide 22

- Add more info about passing arguments into the Scale management methods
- Update sentences order

leestarb committed:
Merge branch 'master' into 423_docs

zevaryx reviewed (approved):
LGtm

AstreaTSS commented on docs/src/Guides/20 Scales.md:
````diff
@@ -203,6 +203,36 @@ bot.grow_scale("Filename_here")
 ```
 
-Finally, for the cool bit of Scales, reloading. Scales allow you to edit your code, and reload it, without restarting the bot.
+Additionally, for the cool bit of Scales, reloading. Scales allow you to edit your code, and reload it, without restarting the bot.
````
"Additionally" here is... confusing. I think "Now" or something similar would be better.

Nitpick though.

silasary commented on docs/src/Guides/20 Scales.md:
```diff
@@ -203,6 +203,36 @@ bot.grow_scale("Filename_here")
+        ...
+
+class AnotherScale(Scale):
+    def __init__(self, _, another_arg: float = 0.0):
```
Why are you using `_` instead of bot here?

LordOfPolls replied:
according to pep this would actually be correct, seeing as you dont use the argument, its purely there because `__new__` requires it to be passed. Though it could be argued as the below code is literally an ellipsis placeholder the true argument name should be used

silasary replied:
For the sake of a readable example, we should definitely use the name.

AstreaTSS commented on docs/src/Guides/20 Scales.md:
```diff
@@ -203,6 +203,36 @@ bot.grow_scale("Filename_here")
 is doing will be abruptly stopped.
+
+
+Finally, you can also pass any **keyword** arguments into module's `setup` and `teardown` functions, it will also send them into your `Scale.shed` method.
```
Am I the only one who got a bit confused by this? I'm *genuinely* asking by the way, it took me a bit to understand but I think that's just me.

LordOfPolls replied:
It's grammatically fragmented. It could be rewritten with clearer wording

silasary replied:
I understand it, but it's definitely not clean

leestarb replied:
How to make it clearer?

LordOfPolls replied:
My suggestion: 
https://github.com/Discord-Snake-Pit/Dis-Snek/pull/424#discussion_r845777544

LordOfPolls commented on docs/src/Guides/20 Scales.md:
```diff
@@ -203,6 +203,36 @@ bot.grow_scale("Filename_here")
 is doing will be abruptly stopped.
+
+
+Finally, you can also pass any **keyword** arguments into module's `setup` and `teardown` functions, it will also send them into your `Scale.shed` method.
```
```suggestion
Any keyword-arguments you pass to the `setup` or `teardown` methods, will also be passed to the `Scale.shed` method. 
```

leestarb committed:
docs: Update Scales guide

by the Lord Of Polls

Co-authored-by: LordOfPolls <22540825+LordOfPolls@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

leestarb committed:
docs: Enhance Scale guide's perception

LordOfPolls commented on docs/src/Guides/20 Scales.md:
````diff
@@ -203,12 +203,13 @@ bot.grow_scale("Filename_here")
 ```
 
-Additionally, for the cool bit of Scales, reloading. Scales allow you to edit your code, and reload it, without restarting the bot.
+Nowadays, for the cool bit of Scales, reloading. Scales allow you to edit your code, and reload it, without restarting the bot.
````
what?

LordOfPolls replied:
I dont think this means what you think it means

leestarb committed:
docs: Fix Scales guide 20
